addons/mail/models/mail_message.py

```diff
@@ -659,9 +659,16 @@ def unlink(self):
         self.mapped('attachment_ids').filtered(
             lambda attach: attach.res_model == self._name and (attach.res_id in self.ids or attach.res_id == 0)
         ).unlink()
+        msg_ids_by_partner = defaultdict(list)
```

Comment:
This could be messages_by_partner and use recordset instead of list.

Reply:
yes I have renamed the variable and used a recordset instead. But I haven’t found a way to avoid a new recordset to be created each time a new element is added which is not the case with a simple list or set. Is it possible ?